test/utils/test_utils.py

```diff
@@ -112,17 +109,15 @@ def elaborate(self, platform):
         return m
 
 
-@parameterized_class(
-    ("name", "size"),
-    [("size" + str(s), s) for s in range(1, 7)],
-)
+@pytest.mark.parametrize("size", range(1, 7))
 class TestCountLeadingZeros(TestCaseWithSimulator):
-    size: int
-
-    def setup_method(self):
+    @pytest.fixture(scope="function", autouse=True)
+    def setup_fixture(self, size):
+        self.size = size
         random.seed(14)
         self.test_number = 40
         self.m = CLZTestCircuit(self.size)
+        yield
```

Comment:
Why there is `yield` added here and not in next test?

Reply:
It turns out it's not needed, when a fixture is not a generator it's called before test runs, which is what we want here. Will remove.